File: 031/31.py
import time

GOAL = 200

time_start = time.time()
combos = 0
for a in range(0, int(GOAL / 1) + 1):
    sa = a
    for b in range(0, int((GOAL - sa) / 2) + 1):
        sb = sa + b * 2
        for c in range(0, int((GOAL - sb) / 5) + 1):
            sc = sb + c * 5
            for d in range(0, int((GOAL - sc) / 10) + 1):
                sd = sc + d * 10
                for e in range(0, int((GOAL - sd) / 20) + 1):
                    se = sd + e * 20
                    for f in range(0, int((GOAL - se) / 50) + 1):
                        sf = se + f * 50
                        for g in range(0, int((GOAL - sf) / 100) + 1):
                            sg = sf + g * 100
                            for h in range(0, int((GOAL - sg) / 200) + 1):
                                sh = sg + h * 200
                                if sh == GOAL:
                                    combos += 1
                                    print (a, b, c, d, e, f, g, h)

print ("Found %d combos in %f seconds" % (combos, time.time() - time_start))

File: 031/31.c
#include <stdio.h>

int main()
{
    int GOAL = 200;

    int combos = 0;
    int a, b, c, d, e, f, g, h;
    int sa, sb, sc, sd, se, sf, sg, sh;
    for (a = 0; a <= GOAL / 1; a++) {
        sa = a;
        for (b = 0; b <= (GOAL - sa) / 2; b++) {
            sb = sa + b * 2;
            for (c = 0; c <= (GOAL - sb) / 5; c++) {
                sc = sb + c * 5;
                for (d = 0; d <= (GOAL - sc) / 10; d++) {
                    sd = sc + d * 10;
                    for (e = 0; e <= (GOAL - sd) / 20; e++) {
                        se = sd + e * 20;
                        for (f = 0; f <= (GOAL - se) / 50; f++) {
                            sf = se + f * 50;
                            for (g = 0; g <= (GOAL - sf) / 100; g++) {
                                sg = sf + g * 100;
                                for (h = 0; h <= (GOAL - sg) / 200; h++) {
                                    sh = sg + h * 200;
                                    if (sh == GOAL) {
                                        combos++;
                                        printf("%d %d %d %d %d %d %d %d\n", a, b, c, d, e, f, g, h);
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    printf("Combos: %d\n", combos);
    return 0;
}
